sympy/physics/mechanics/body.py

```diff
@@ -133,8 +135,12 @@ def __init__(self, name, masscenter=None, mass=None, frame=None,
             self.frame = frame
             self.masscenter = masscenter
             Particle.__init__(self, name, masscenter, _mass)
+            self.is_rigidbody = False
+            self.is_particle = True
         else:
             RigidBody.__init__(self, name, masscenter, frame, _mass, _inertia)
+            self.is_rigidbody = True
+            self.is_particle = False
```

Comment:
Can't a simple method (or even property) be:

```
def is_rigidbody(self):
    if self.inertia is not None:
        return True
    else:
        return False
```

?

Reply:
1. Changes should be consisted in `RigidBody`, `Body` and `Particle`.
2. In that line we have already differentiated that `Body` is rigidbody or body, so I think we don't need extra property or anything for that.

Reply:
My suggestion is to make the property detect if it is a particle or rigidbody, then the answer is guaranteed to be correct.

Reply:
In `Body.__init__` we are detecting if it is a particle or rigidbody so I don't think we need a separate property for that.

Reply:
I'm suggesting removing the check in `__init__` and putting the check in the property (or method).

Reply:
The correct way seems better because we need to instantiate `Particle` or `RigidBody` and that is to be done in `__init__`.